io/src/main/scala/fs2/io/file/Files.scala

```diff
@@ -0,0 +1,510 @@
+
+  /**
+    * Like `delete`, but will not fail when the path doesn't exist.
+    */
+  def deleteIfExists(path: Path): F[Boolean]
+
+  /**
+    * Recursively delete a directory
+    */
+  def deleteDirectoryRecursively(path: Path, options: Set[FileVisitOption] = Set.empty): F[Unit]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory.
+    */
+  def directoryStream(path: Path): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory, filtering the results by the given predicate.
+    */
+  def directoryStream(path: Path, filter: Path => Boolean): Stream[F, Path]
+
+  /**
+    * Creates a stream of [[Path]]s inside a directory which match the given glob.
+    */
+  def directoryStream(path: Path, glob: String): Stream[F, Path]
+
+  /**
+    * Checks if a file exists.
+    *
+    * Note that the result of this method is immediately outdated. If this
+    * method indicates the file exists then there is no guarantee that a
+    * subsequence access will succeed. Care should be taken when using this
+    * method in security sensitive applications.
+    */
+  def exists(path: Path, flags: Seq[LinkOption] = Seq.empty): F[Boolean]
+
+  /**
+    * Moves (or renames) a file from the source to the target path.
+    *
+    * By default, the move fails if the target file already exists or is a symbolic link.
+    */
+  def move(source: Path, target: Path, flags: Seq[CopyOption] = Seq.empty): F[Path]
+
+  /** Creates a `FileHandle` for the file at the supplied `Path`. */
+  def openPath(path: Path, flags: Seq[OpenOption]): Resource[F, FileHandle[F]]
+
+  /** Creates a `FileHandle` for the supplied `FileChannel`. */
+  def openFileChannel(channel: F[FileChannel]): Resource[F, FileHandle[F]]
+
+  /**
+    * Get file permissions as set of [[PosixFilePermission]].
+    *
+    * Note: this will only work for POSIX supporting file systems.
+    */
+  def permissions(path: Path, flags: Seq[LinkOption] = Seq.empty): F[Set[PosixFilePermission]]
+
+  /**
+    * Reads all data from the file at the specified `java.nio.file.Path`.
+    */
+  def readAll(path: Path, chunkSize: Int): Stream[F, Byte]
+
+  /**
+    * Reads a range of data synchronously from the file at the specified `java.nio.file.Path`.
+    * `start` is inclusive, `end` is exclusive, so when `start` is 0 and `end` is 2,
+    * two bytes are read.
+    */
+  def readRange(path: Path, chunkSize: Int, start: Long, end: Long): Stream[F, Byte]
+
+  /**
+    * Set file permissions from set of [[PosixFilePermission]].
+    *
+    * Note: this will only work for POSIX supporting file systems.
+    */
+  def setPermissions(path: Path, permissions: Set[PosixFilePermission]): F[Path]
+
+  /**
+    * Returns the size of a file (in bytes).
+    */
+  def size(path: Path): F[Long]
+
+  /**
+    * Creates a stream containing the path of a temporary file.
+    *
+    * The temporary file is removed when the stream completes.
+    */
+  def tempFileStream(
```

Comment:
So, the implementation for this funciton, below, is defined as `Stream.resource` applied to the function below. 

- On one hand, since the `Stream.resource` method does not use the `Sync[F]` instance, this implementation could be added here, on the top-level definition. 
- On the other hand, perhaps there is no need to add a method for what is a fixed combination of a primitive and other method in this trait.

Reply:
Yeah, not sure why we have both. Let's keep just the `Resource` returning versions.

Reply:
👍 for only keeping the `=> Resource` version